yield_plugin/src/yield_plugin.cpp

```diff
@@ -65,6 +65,11 @@ namespace yield_plugin
     return fabs(get_trajectory_end_time(trajectory) - get_trajectory_start_time(trajectory));
   }
 
+  double get_predicted_steps_duration(const std::vector<carma_perception_msgs::msg::PredictedState>& trajectory)
```

Comment:
Optional: Consider renaming to `get_trajectory_duration()` to be more general and more self-explanatory. `PredictedStates` (or `_predicted_steps_`) seems to be an implementation detail.